server/src/main/java/org/opensearch/rest/action/search/RestSearchAction.java

```diff
@@ -235,13 +236,12 @@ private static void parseSearchSource(final SearchSourceBuilder searchSourceBuil
             searchSourceBuilder.query(queryBuilder);
         }
 
-        int from = request.paramAsInt("from", -1);
-        if (from != -1) {
-            searchSourceBuilder.from(from);
+        if (request.hasParam("from")) {
+            searchSourceBuilder.from(request.paramAsInt("from", SearchService.DEFAULT_FROM));
```

Comment:
Thanks for adding that back! This change looks good to me, but it would be nice not to have to pass the second param if it will never be utilized. The default will never be chosen because its already wrapped in a block that checks if the request contains the param.

Reply:
We don't have a overloaded `paramAsInt` method which omits the default value, do you think we could add one for this case?

Reply:
@gaobinlong Yes, but you can make this as part of another PR.